src/Toolkit/V.php

```diff
@@ -239,13 +239,45 @@ public static function __callStatic(string $method, array $arguments): bool
     },
 
     /**
-     * Checks for a valid date
-     */
-    'date' => function ($value): bool {
-        $date = date_parse($value);
-        return $date !== false &&
-                $date['error_count'] === 0 &&
-                $date['warning_count'] === 0;
+     * Checks for a valid date and is able to compare
+     * dates against each other.
+     *
+     * Pass only the first argument to check for a valid date.
+     * Pass an operator as second argument and another date as
+     * third argument to compare them.
+     */
+    'date' => function ($value, string $operator = null, $test = null): bool {
```

Comment:
Both `date_parse()` and `strtotime()` only support strings as arguments, so we can be strict here like in some other validators:

```suggestion
    'date' => function (string $value, string $operator = null, string $test = null): bool {
```